File: Semester_3/OOP/week2/q3.java
import java.util.Scanner;

class q3
{
	static void Lsearch(int arr[], int n, int ele )
	{
		int i, flag=0; 
		for(i = 0; i<n; i++)
		{
			if(arr[i] == ele)
			{
				//printf("%d", i);
				flag = 1;
				break;
			}
		}
		if(flag == 1)
			System.out.println("Element found at position" +  i);
		else 
			System.out.printf("Element not found");
	}

	public static void main(String[] args)
	{
		int n, i, ele;
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the number of elements: ");
		n = sc.nextInt();
		int[] arr = new int[n];
		System.out.println("Enter the elements of the array: \n");
		for(i = 0; i<n; i++)
		{
			arr[i] = sc.nextInt();
		}
		System.out.println("Enter the element to be found: ");
		ele = sc.nextInt();
		Lsearch(arr, n, ele);
	}
}

File: Semester_6/PCAP_lab/lab4/q3.c
/*Write a MPI program to read a 3X3 matrix. Enter an element to be searched in the root 
process. Find the number of occurrences of this element in the matrix using three processes. 
Also, handle different errors using error handling routines. */

#include <stdio.h>
#include <mpi.h>
 
int main (int a,char *b[]) 
{
	int rank, ele,size;
	int A[3][3], B[3], count=0, tc = 0;
	MPI_Init(&a, &b);
	MPI_Comm_rank(MPI_COMM_WORLD, &rank);
	MPI_Comm_size(MPI_COMM_WORLD, &size);
	MPI_Errhandler_set(MPI_COMM_WORLD,MPI_ERRORS_RETURN);

	int error; 
	error=MPI_Bcast(&size, 1, MPI_INT, 4, MPI_COMM_WORLD);
	if (error != MPI_SUCCESS) 
	{
		char s[100];
		int len, class1;
		MPI_Error_string(error, s, &len);
		MPI_Error_class(error, &class1);
		fprintf(stderr, "Error description is %s", s);
		fflush(stderr);
		fprintf(stderr, "Error class is %d", class1);
		fflush(stderr);
	}
	    	
	if (rank == 0)
	{
		printf("Enter elements into matrix: \n");
		for (int i = 0; i < 3; i++)
			for (int j = 0; j < 3; j++)
				scanf("%d", &A[i][j]);
		        
		printf("Enter the element to count: \n");
		scanf("%d", &ele);
	}
	MPI_Bcast(&ele, 1, MPI_INT, 0, MPI_COMM_WORLD);
	MPI_Scatter(A , 3, MPI_INT, B, 3, MPI_INT, 0, MPI_COMM_WORLD);
	 
	for (int i = 0; i < 3; i++) 
	{
		printf("%d ",B[i]);
		if (B[i] == ele) 
		{
			count ++;
		}
	}
	printf("\nProcess %d found %d occurrences\n",rank,count);
	MPI_Reduce(&count, &tc, 1, MPI_INT, MPI_SUM, 0, MPI_COMM_WORLD);
	if (rank == 0)
	{
		printf("\nNumber of occurrences is %d\n",tc);
	}
	MPI_Finalize();
	return 0; 
}